asserts/pool.go

```diff
@@ -314,14 +440,30 @@ func (p *Pool) addUnresolved(unresolved *AtRevision, gnum uint16) error {
 		return nil
 	}
 	uniq := unresolved.Ref.Unique()
-	var u *unresolvedRec
+	var u unresolvedAssertRecord
 	if u = p.unresolved[uniq]; u == nil {
 		u = &unresolvedRec{
 			at: unresolved,
 		}
 		p.unresolved[uniq] = u
 	}
-	u.merge(unresolved, gnum, p.groupings)
+
+	urec := u.(*unresolvedRec)
+	urec.merge(unresolved, gnum, p.groupings)
+	return nil
+}
+
+func (p *Pool) addUnresolvedSeq(unresolved *AtSequence, gnum uint16) error {
```

Comment:
for our use case likely we will not ask at different point for the same unresolved sequence, because as we discussed they are never prereqs.  It might still worth leaving a TODO about adding a check in the future though. To be clear that wouldn't be about prereq.  But repeated calls to AddToUpdateSequence or AddUnresolvedSeq about the same sequence.